### ISSUE TITLE
Command prefixes can clash

### ISSUE THREAD

**Issue by AndrewFerr** [matrix-org/matrix-hookshot#421]
**Describe the bug**
When there are multiple connections in the same room, it is possible for them to share the same command prefix (`!gl`, `!gh`, etc).  If a command with a shared prefix is run, it will only be processed against one of the connections.  This prevents being able to run commands for some of the connections that share a command prefix.

**To Reproduce**
1. Connect two GitLab projects to a room, and do not customize the command prefix from its default of `!gl`
2. Try to run `!gl` commands in that room
3. Notice that only one of the projects accepts commands

**Expected behavior**
It should be forbidden for multiple connections to share the same command prefix.  If adding a new connection with a default prefix clashes with one already in use within a room, either the prefix should be modified to be unique, or the action of adding a connection should require the user to set a custom, unique prefix for it.

**Screenshots**
N/A

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Additional context**
N/A

**Comment by HarHarLinks** [comment]
see also #244: currently command prefix are confusing anyway

**Comment by Half-Shot** [comment]
The workaround at the moment is adding your own prefix, though I would like to see the ConnectionManager basically handle this automatically. I like the idea of the connection manager handling prefixes where:

Connections with matching prefixes automatically gain a identifier prefix like `!gl` -> `!gl my-org/my-project` and using `!gl` will just warn you to use the right one. It would still be nice to set a default though, as in lots of places we tend to expect one. This could be the case of adding some state to the connection to denote that it is the default one.

**Comment by AndrewFerr** [comment]
Even after #540, another issue remains: adding connections via bot commands (instead of the widget or provisioning API) allows creating connections with a prefix that's already in use.

Part of the problem is that bot commands don't allow setting a custom prefix for a new connection.
